core/cache/multiLevelBackend.go

```diff
@@ -0,0 +1,128 @@
+package cache
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"flamingo.me/flamingo/v3/framework/flamingo"
+)
+
+type (
+	// MultiLevelBackend instance representation
+	MultiLevelBackend struct {
+		backends []Backend
+		logger   flamingo.Logger
+	}
+	// MultiLevelBackendOptions representation
+	MultiLevelBackendOptions struct {
+		Backends []Backend
+	}
+)
+
+// NewMultiLevelBackend creates a MultiLevelBackend isntance
+func NewMultiLevelBackend(options MultiLevelBackendOptions, logger flamingo.Logger) *MultiLevelBackend {
+	return &MultiLevelBackend{
+		backends: options.Backends,
+		logger:   logger,
+	}
+}
+
+// Inject MultiLevelBackend dependencies
+func (mb *MultiLevelBackend) Inject(logger flamingo.Logger) {
+	mb.logger = logger
+}
+
+// Get entry by key
+func (mb *MultiLevelBackend) Get(key string) (entry *Entry, found bool) {
+	for _, backend := range mb.backends {
+		entry, found := backend.Get(key)
+		if found {
+			return entry, found
```

Comment:
You set key/value on all backends but get it only from one. Isn't this potentially inconsistent?

Reply:
yes, that's the wanted behavior. both caches should always have the newest entry-version. but we take entry of the first cache which contains the key. in most cases this should be the first cache. only in scalup-cenarios the first cache will be empty and the second (mostly shared) cache will have the entry.